### PR TITLE
fix(mobile): mobile logging out randomly

### LINKED ISSUE #2980: mobile: should be more resilient to temporary communication issues
### The bug

I use Tailscale to access Immich, and have an iOS automation which connects to Tailscale when I open the app. I found that Immich will try to resolve the host, fail, and then assume I'm actually unauthenticated and ask me to login again. It would be nice to:
1. Not make that assumption.
2. Handle transient connection failures more gracefully.

### The OS that Immich Server is running on

N/A

### Version of Immich Server

N/A

### Version of Immich Mobile App

1.63.0 build.103

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
N/A
```

### Your .env content

```Shell
N/A
```

### Reproduction steps

```bash
N/A
```

### Additional information

_No response_

### LINKED ISSUE #8146: Immich iOS keeps logging out after a day or so
### The bug

I recently switched from exposing a port on my network to using Cloudflare tunnels to access Immich over the web. Works great, however I’ve noticed that I keep getting logged out after some time. 

Sometimes it’s a day, other times a couple. Seems to be when I’m off local internet for some time. I read that some network resiliency was added a while back but not sure if it’s entirely functional. Cloudflare tunnel doesn’t report to be unreachable at any given time.

### The OS that Immich Server is running on

unRAID 6.12 - Docker

### Version of Immich Server

1.98.1

### Version of Immich Mobile App

1.99.0 b 145

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
version: "3.8"

services:
  immich-server:
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    command: [ "start.sh", "immich" ]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
    ports:
      - 2283:3001
    env_file:
      - .env
    depends_on:
      - redis
      - database
    restart: always

  immich-microservices:
    container_name: immich_microservices
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    # extends:
    #   file: hwaccel.yml
    #   service: hwaccel
    command: [ "start.sh", "microservices" ]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
    env_file:
      - .env
    depends_on:
      - redis
      - database
    restart: always

  immich-machine-learning:
    container_name: immich_machine_learning
    image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}
    volumes:
      - model-cache:/cache
    env_file:
      - .env
    restart: always

  redis:
    container_name: immich_redis
    image: redis:6.2-alpine@sha256:70a7a5b641117670beae0d80658430853896b5ef269ccf00d1827427e3263fa3
    restart: always

  database:
    container_name: immich_postgres
    image: registry.hub.docker.com/tensorchord/pgvecto-rs:pg14-v0.2.0@sha256:90724186f0a3517cf6914295b5ab410db9ce23190a2d9d0b9dd6463e3fa298f0
    env_file:
      - .env
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
    volumes:
      - pgdata:/var/lib/postgresql/data
    restart: always

volumes:
  pgdata:
  model-cache:
  tsdata:
```

### Your .env content

```Shell
# You can find documentation for all the supported env variables at https://immich.app/docs/install/environment-variables

# The location where your uploaded files are stored
UPLOAD_LOCATION=./immich-data

# The Immich version to use. You can pin this to a specific version like "v1.71.0"
IMMICH_VERSION=release

# Connection secrets for postgres and typesense. You should change these to random passwords
TYPESENSE_API_KEY=key
DB_PASSWORD=passwordb

# The values below this line do not need to be changed
###################################################################################
DB_HOSTNAME=immich_postgres
DB_USERNAME=username
DB_DATABASE_NAME=dbname

REDIS_HOSTNAME=immich_redis
```

### Reproduction steps

```bash
1. Login at home
2. Works fine for some time
3. Come back randomly a few days later and I’m logged out. Rest of family also is
...
```

### Additional information

_No response_

### LINKED ISSUE #9745: App can't hit .well_known, hits incorrect API endpoint, then logs out
### The bug

```
> 172.24.0.7 - - [21/May/2024:21:36:52 +0000] "GET /api/server-info HTTP/1.1" 404 131 "-" "Dart/3.3 (dart:io)" "174.206.38.109" "<domain>" "/api/server-info" "-" "-"
> 172.24.0.7 - - [21/May/2024:21:36:52 +0000] "GET /api/asset/device/bf001c7874f972a4ddfceb3774bdce8261c269ce025ed8fe3882800fbc68e5bf HTTP/1.1" 200 36458 "-" "Dart/3.3 (dart:io)" "174.206.38.109" "<domain>" "/api/asset/device/bf001c7874f972a4ddfceb3774bdce8261c269ce025ed8fe3882800fbc68e5bf" "immich" "192.168.176.3:3001"
> 172.24.0.7 - - [21/May/2024:21:36:55 +0000] "GET /api/socket.io/?EIO=4&transport=websocket HTTP/1.1" 101 362 "-" "Dart/3.3 (dart:io)" "174.206.38.109" "<domain>" "/api/socket.io/?EIO=4&transport=websocket" "immich" "192.168.176.3:3001"
> 172.24.0.7 - - [22/May/2024:04:00:36 +0000] "GET /api/server-info/ping HTTP/1.1" 200 14 "-" "Dart/3.3 (dart:io)" "<ip>" "<domain>" "/api/server-info/ping" "immich" "192.168.176.3:3001"
> 172.24.0.7 - - [22/May/2024:04:00:36 +0000] "GET /.well-known/immich HTTP/1.1" 200 56 "-" "Dart/3.3 (dart:io)" "<ip>" "<domain>" "/.well-known/immich" "immich" "192.168.176.3:3001"
> 172.24.0.7 - - [22/May/2024:05:00:42 +0000] "GET /user/me HTTP/1.1" 204 0 "-" "Dart/3.3 (dart:io)" "<ip>" "<domain>" "/user/me" "-" "-"
> 172.24.0.7 - - [22/May/2024:05:00:42 +0000] "POST /auth/logout HTTP/1.1" 204 0 "-" "Dart/3.3 (dart:io)" "<ip>" "<domain>" "/auth/logout" "-" "-"
```
`204` response is due to my reverse proxy which is set to not send `404`, `503` etc..

My `/.well-known/immich` response:

```
{
     "api": {
         "endpoint": "/api"
    }
}
```

image

### The OS that Immich Server is running on

Ubuntu 20.04

### Version of Immich Server

v1.105

### Version of Immich Mobile App

v1.105.1

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
N/A
```

### Your .env content

```Shell
N/A
```

### Reproduction steps

```bash
1.
2.
3.
...
```

### Relevant log output

_No response_

### Additional information

Discord discussion: https://discord.com/channels/979116623879368755/1049703391762321418

### PR DESCRIPTION
**Pull request by alextran1502**
Fixes #2980 
Fixes #9745
Fixes #8146

This PR simplifies and fixes the issue of occasionally logging out the user when the `.well-know` endpoint or the getMyUser endpoint returns a status code other than `503` when it cannot be reached. We should now always login provided we have a local version of the user (one successful login has occurred previously) and the API doesn't return a 401 (Auth token has been cancelled)

### CODE REVIEW

**Review by zackpollard — APPROVED** [review]

**Comment by zackpollard** [comment]
Approved. But also I wrote a lot of the code so let's get at least one review from the mobile guys 🙂

**Review by jrasm91 — APPROVED** [review]

File: mobile/lib/providers/authentication.provider.dart [hunk 1]
```diff
@@ -165,57 +164,56 @@ class AuthenticationNotifier extends StateNotifier<AuthenticationState> {
...
-        if (error.innerException is SocketException) {
-          state = state.copyWith(isAuthenticated: true);
-        }
-      } catch (error, stackTrace) {
-        _log.severe(
-          "Error getting user information from the server [CATCH ALL]",
-          error,
-          stackTrace,
-        );
+    UserAdminResponseDto? userResponse;
```
Comment (martyfuhry) [inline mobile/lib/providers/authentication.provider.dart:169]: Make final

File: mobile/lib/providers/authentication.provider.dart [hunk 2]
```diff
@@ -165,57 +164,56 @@ class AuthenticationNotifier extends StateNotifier<AuthenticationState> {
...
+    UserPreferencesResponseDto? userPreferences;
```
Comment (martyfuhry) [inline mobile/lib/providers/authentication.provider.dart:170]: Final

File: mobile/lib/providers/authentication.provider.dart [hunk 3]
```diff
@@ -165,57 +164,56 @@ class AuthenticationNotifier extends StateNotifier<AuthenticationState> {
...
+    try {
+      var responses = await Future.wait([
```
Comment (martyfuhry) [inline mobile/lib/providers/authentication.provider.dart:172]: Final. Is there a good reason to do these in a Future.wait block?

File: mobile/lib/providers/authentication.provider.dart [hunk 4]
```diff
@@ -165,57 +164,56 @@ class AuthenticationNotifier extends StateNotifier<AuthenticationState> {
...
+        _apiService.usersApi.getMyUser(),
+        _apiService.usersApi.getMyPreferences(),
+      ]);
+      userResponse = responses[0] as UserAdminResponseDto;
+      userPreferences = responses[1] as UserPreferencesResponseDto;
+    } on ApiException catch (error, stackTrace) {
```
Comment (martyfuhry) [inline mobile/lib/providers/authentication.provider.dart:178]: How does Future.wait handle exceptions? Will this be thrown in this manner?

File: mobile/lib/pages/common/splash_screen.page.dart [hunk 5]
```diff
@@ -19,45 +18,22 @@ class SplashScreenPage extends HookConsumerWidget {
   Widget build(BuildContext context, WidgetRef ref) {
     final apiService = ref.watch(apiServiceProvider);
     final serverUrl = Store.tryGet(StoreKey.serverUrl);
+    final endpoint = Store.tryGet(StoreKey.serverEndpoint);
     final accessToken = Store.tryGet(StoreKey.accessToken);
     final log = Logger("SplashScreenPage");
 
     void performLoggingIn() async {
```
Comment (martyfuhry) [inline mobile/lib/pages/common/splash_screen.page.dart:25]: Is there any way to put this function somewhere else? I typically don't expect to find lots of business logic in the build function of a widget. Maybe we can add this function to a relevant provider. The ui should really just be watching the state of the notifier and updating accordingly.

**Review by martyfuhry — DISMISSED** [review]
I don't really follow the logic of how this improves the random logouts, but I'm reviewing on my phone without adequate coffee haha! It looks very good and clean. I would like to see some more refactoring. Perhaps some comments about where *specifically* in this code the random logouts are fixed?

Thank you, I hope my review helps.

File: mobile/lib/providers/authentication.provider.dart (hunk 3 above)
Comment (zackpollard) [reply mobile/lib/providers/authentication.provider.dart:172]: It'll make the requests in parallel, just a performance optimisation

File: mobile/lib/providers/authentication.provider.dart (hunk 4 above)
Comment (zackpollard) [reply mobile/lib/providers/authentication.provider.dart:178]: Any exceptions will be presented in the same way. When the await for the Future.wait resolves or rejects

File: mobile/lib/pages/common/splash_screen.page.dart (hunk 5 above)
Comment (zackpollard) [reply mobile/lib/pages/common/splash_screen.page.dart:25]: I appreciate that, but also this is the current state, we are trying to fix the problem without creating more problems by refactoring stuff too much.

File: mobile/lib/providers/authentication.provider.dart (hunk 3 above)
Comment (alextran1502) [reply mobile/lib/providers/authentication.provider.dart]: these variables will be re-assigned and checked for null.

File: mobile/lib/providers/authentication.provider.dart (hunk 2 above)
Comment (alextran1502) [reply mobile/lib/providers/authentication.provider.dart:170]: These variables will be re-assigned and checked for null.

File: mobile/lib/providers/authentication.provider.dart (hunk 1 above)
Comment (alextran1502) [reply mobile/lib/providers/authentication.provider.dart:169]: This variable will be re-assigned and checked for null.

**Comment by krisavi** [comment]
I am wondering if that change caused #11518.

**Comment by Krankenkasse** [comment]
I still get the requests to  _"/api/.well-known/immich"_


```
[07/Feb/2025:18:23:24 +0100] - 200 200 - GET https XXX.de "/api/server/ping" [Client XXX] [Length 14] [Gzip -] [Sent-to 192.168.178.xxx] "Dart/3.5 (dart:io)" "-"
[07/Feb/2025:18:23:24 +0100] - 200 200 - GET https XXX.de "/api/server/ping" [Client XXX] [Length 14] [Gzip -] [Sent-to 192.168.178.xxx] "Dart/3.5 (dart:io)" "-"
[07/Feb/2025:18:23:24 +0100] - 404 404 - GET https XXX.de "/api/.well-known/immich" [Client XXX] [Length 112] [Gzip -] [Sent-to 192.168.178.xxx] "Dart/3.5 (dart:io)" "-"
[07/Feb/2025:18:23:24 +0100] - 404 404 - GET https XXX.de "/api/.well-known/immich" [Client XXX] [Length 112] [Gzip -] [Sent-to 192.168.178.xxx] "Dart/3.5 (dart:io)" "-"
[07/Feb/2025:18:23:24 +0100] - 200 200 - GET https XXX.de "/api/users/me" [Client XXX] [Length 469] [Gzip -] [Sent-to 192.168.178.xxx] "Dart/3.5 (dart:io)" "-"
``` 

Thats the requests that are comming from the app when doing a "cold start" of it.

the first log in the app is Failed to fetch users at /api/users

The problem is that my reverse proxy fail to ban banned the client for receiving to many 404

App Version 1.125.5 build.181
Server Version 1.125.7

**Pull request merged**